src/backend/distributed/planner/relation_restriction_equivalence.c

```diff
@@ -1778,19 +1800,116 @@ RelationRestrictionPartitionKeyIndex(RelationRestriction *relationRestriction)
 		partitionKeyTargetAttrIndex++;
 
 		if (!targetEntry->resjunk &&
-			IsA(targetExpression, Var) &&
-			IsPartitionColumn(targetExpression, relationPlannerParseQuery))
+			IsA(targetExpression, Var))
 		{
 			Var *targetColumn = (Var *) targetExpression;
 
-			if (targetColumn->varno == relationRestriction->index)
+			if (!IsPartitionColumn(targetExpression, relationPlannerParseQuery))
+			{
+				continue;
+			}
+
+			RangeTblEntry *resRTE = FindRelationRTEForColumn(
```

Comment:
Do we really need `FindRelationRTEForColumn`? Wouldn't `FindReferencedTableColumn` find the target column even if it is coming from a JOIN?

Reply:
No we don't need it, you are right. I changed the ordering of calls to avoid calling `FindRelationRTEForColumn`.